### ISSUE TITLE
Incorrect documentation

### ISSUE THREAD

**Issue by Imperat** [shogowada/json-rpc-2.0#10]
Hi, @shogowada 
I read the documentation and I have missunderstood the section "With authentication".

Your code example is:
```
const server = new JSONRPCServer();

// If the method is a higher-order function (a function that returns a function),
// it will pass the custom parameter to the returned function.
// Use this to inject whatever information that method needs outside the regular JSON-RPC request.
server.addMethod("echo", ({ text }) => ({ userID }) => `${userID} said ${text}`);

app.post("/json-rpc", (req, res) => {
  const jsonRPCRequest = req.body;
  const userID = getUserID(req);

  // server.receive takes an optional second parameter.
  // The parameter will be injected to the JSON-RPC method if it was a higher-order function.
  server.receive(jsonRPCRequest, { userID }).then(jsonRPCResponse => {
    if (jsonRPCResponse) {
      res.json(jsonRPCResponse);
    } else {
      res.sendStatus(204);
    }
  });
});

const getUserID = (req) => // Do whatever to get user ID out of the request
```

But according TS definitions I see that `server.addMethod` takes a value of type:
 `export declare type SimpleJSONRPCMethod = (params?: Partial<JSONRPCParams>) => any;`

Did you mean that we need to use `addMethodAdvanced` instead? I see it has the correct type definition.

**Comment by shogowada** [comment]
It is intentional.

The type of `SimpleJSONRPCMethod` practically is:
`(params: JSONRPCParams) => Result | (serverParams: ServerParams) => Result`

but given `Result` is `any`, it becomes:
`(params: JSONRPCParams) => any | (serverParams: ServerParams) => any`

Since `any | (serverParams: ServerParams) => any` is `any`, it becomes:
`(params: JSONRPCParams) => any`

I believe I saw `any | Something` should always be `any` in TypeScript doc (although I cannot find it at the moment), so I made it `(params: JSONRPCParams) => any`.

`addMethodAdvanced` is intended to handle JSON-RPC request/notification as is and return a complete JSON-RPC response (we already know what `JSONRPCResponse` looks like, hence no `any`), while `addMethod` will only give you `params` part of the JSON-RPC request/notification and is supposed to only return the `result` part of the JSON-RPC response.

Let me know if this makes sense 🙂
